tcpdialer.go

```diff
@@ -327,7 +327,11 @@ func (d *TCPDialer) tryDial(network string, addr *net.TCPAddr, deadline time.Tim
 		defer func() { <-concurrencyCh }()
 	}
 
-	dialer := net.Dialer{LocalAddr: d.LocalAddr}
+	dialer := net.Dialer{}
+	if d.LocalAddr != nil {
+		dialer = net.Dialer{LocalAddr: d.LocalAddr}
```

Comment:
You are right, my mistake. But please simplify the code a bit to:
```suggestion
		dialer.LocalAddr = d.LocalAddr
```

Reply:
Sure, I'll close this PR and file another one, it's my bad to delete the forked repo accidentally :-)